x-pack/plugins/apm/server/lib/index_pattern/create_static_index_pattern.ts

```diff
@@ -35,8 +45,31 @@ export async function createStaticIndexPattern(
       return false;
     }
 
+    const apmIndexPatternTitle = getApmIndexPatternTitle(config);
+
+    if (!overwrite) {
+      try {
+        const {
+          attributes: { title: existingApmIndexPatternTitle },
+        }: {
+          attributes: ApmIndexPatternAttributes;
+        } = await savedObjectsClient.get(
+          'index-pattern',
+          APM_STATIC_INDEX_PATTERN_ID
+        );
+        // if the existing index pattern does not matches the new one, force an update
+        if (existingApmIndexPatternTitle !== apmIndexPatternTitle) {
+          overwrite = true;
+        }
```

Comment:
Does this affect anything? you received `overwrite`  as a property and used it on line 50, what happens when you change to true?

Reply:
arguments are mutable, this includes destructured args. Is there a better way to do this?

Reply:
but the `overwrite` is never used after you changed it to `true`, right? is it saved for the next time this function is called?

Reply:
it is used further down in the savedObjectClient.create call: https://github.com/elastic/kibana/blob/c89b6469d0cde931ec42489f260a9302c342ca92/x-pack/plugins/apm/server/lib/index_pattern/create_static_index_pattern.ts#L82

Reply:
ah got it, I didn't see it